hw/mcu/nxp/kinetis/src/hal_qspi.c

```diff
@@ -0,0 +1,410 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <string.h>
+#include <assert.h>
+#include <stdint.h>
+#include "os/mynewt.h"
+#if MYNEWT_VAL(QSPI_ENABLE)
+#include <mcu/cmsis_nvic.h>
+#include <hal/hal_flash_int.h>
+
+#include <fsl_port.h>
+#include <fsl_qspi.h>
+#include <fsl_clock.h>
+
+#if MYNEWT_VAL(QSPI_FLASH_SECTOR_SIZE) < 1
+#error QSPI_FLASH_SECTOR_SIZE must be set to the correct value in bsp syscfg.yml
+#endif
+
+#if MYNEWT_VAL(QSPI_FLASH_SECTOR_COUNT) < 1
+#error QSPI_FLASH_SECTOR_COUNT must be set to the correct value in bsp syscfg.yml
+#endif
+
+/* #if MYNEWT_VAL(QSPI_PIN_CS) < 0 */
+/* #error QSPI_PIN_CS must be set to the correct value in bsp syscfg.yml */
+/* #endif */
+
+/* #if MYNEWT_VAL(QSPI_PIN_SCK) < 0 */
+/* #error QSPI_PIN_SCK must be set to the correct value in bsp syscfg.yml */
+/* #endif */
+
+/* #if MYNEWT_VAL(QSPI_PIN_DIO0) < 0 */
+/* #error QSPI_PIN_DIO0 must be set to the correct value in bsp syscfg.yml */
+/* #endif */
+
+/* #if MYNEWT_VAL(QSPI_PIN_DIO1) < 0 */
+/* #error QSPI_PIN_DIO1 must be set to the correct value in bsp syscfg.yml */
+/* #endif */
+
+/* #if MYNEWT_VAL(QSPI_PIN_DIO2) < 0 && (MYNEWT_VAL(QSPI_READOC) > 2 || MYNEWT_VAL(QSPI_WRITEOC) > 1) */
+/* #error QSPI_PIN_DIO2 must be set to the correct value in bsp syscfg.yml */
+/* #endif */
+
+/* #if MYNEWT_VAL(QSPI_PIN_DIO3) < 0 && (MYNEWT_VAL(QSPI_READOC) > 2 || MYNEWT_VAL(QSPI_WRITEOC) > 1) */
+/* #error QSPI_PIN_DIO3 must be set to the correct value in bsp syscfg.yml */
+/* #endif */
```

Comment:
Should all those commented out lines be removed?

Reply:
Addressed.